app/src/main/java/org/jellyfin/androidtv/playback/overlay/CustomPlaybackTransportControlGlue.java

```diff
@@ -0,0 +1,239 @@
+        closedCaptionsAction = new ClosedCaptionsAction(context, this);
+        adjustAudioDelayAction = new AdjustAudioDelayAction(context, this);
+        zoomAction = new ZoomAction(context, this);
+        chapterAction = new ChapterAction(context, this);
+
+        previousLiveTvChannelAction = new PreviousLiveTvChannelAction(context, this);
+        channelBarChannelAction = new ChannelBarChannelAction(context, this);
+        guideAction = new GuideAction(context, this);
+        recordAction = new RecordAction(context, this);
+    }
+
+    @Override
+    protected void onCreatePrimaryActions(ArrayObjectAdapter primaryActionsAdapter) {
+        this.primaryActionsAdapter = primaryActionsAdapter;
+
+        primaryActionsAdapter.add(playPauseAction);
+
+        if (canSeek()) {
+            primaryActionsAdapter.add(rewindAction);
+            primaryActionsAdapter.add(fastForwardAction);
+        }
+
+        if (hasSubs()) {
+            primaryActionsAdapter.add(closedCaptionsAction);
+        }
+
+        if (hasMultiAudio()) {
+            primaryActionsAdapter.add(selectAudioAction);
+        }
+
+        if (isLiveTv()) {
+            primaryActionsAdapter.add(channelBarChannelAction);
+            primaryActionsAdapter.add(guideAction);
+        }
+    }
+
+    @Override
+    protected void onCreateSecondaryActions(ArrayObjectAdapter secondaryActionsAdapter) {
+        this.secondaryActionsAdapter = secondaryActionsAdapter;
+
+        if (isLiveTv()) {
+            secondaryActionsAdapter.add(previousLiveTvChannelAction);
+            if (canRecordLiveTv()) {
+                secondaryActionsAdapter.add(recordAction);
+                recordingStateChanged();
+            }
+        }
+
+        if (hasNextItem()) {
+            secondaryActionsAdapter.add(skipNextAction);
+        }
+
+
+        if (hasChapters()) {
+            secondaryActionsAdapter.add(chapterAction);
+        }
+
+        if (!isNativeMode()) {
+            secondaryActionsAdapter.add(adjustAudioDelayAction);
+        } else {
+            secondaryActionsAdapter.add(zoomAction);
+        }
+    }
+
+    @Override
+    public void onActionClicked(Action action) {
+        if (action == playPauseAction) {
+            if (playPauseAction.getIndex() == PlaybackControlsRow.PlayPauseAction.INDEX_PAUSE) {
+                getPlayerAdapter().pause();
+            } else if (playPauseAction.getIndex() == PlaybackControlsRow.PlayPauseAction.INDEX_PLAY) {
+                getPlayerAdapter().play();
+            }
+            playPauseAction.nextIndex();
+        } else if (action == rewindAction) {
+            getPlayerAdapter().rewind();
+        } else if (action == fastForwardAction) {
+            getPlayerAdapter().fastForward();
+        } else if (action == skipNextAction) {
+            getPlayerAdapter().next();
+        }
+        notifyActionChanged(action);
+    }
+
+    public void onCustomActionClicked(Action action, View view) {
```

Comment:
See my previous comment on the way these handlers are implemented. Would be better if you could just do `action.onClicked();` or something like that

Reply:
Good point!
But then I will have to keep an action class for every action I add.